### DISCUSSION TITLE
Can a source generator reference itself as an analyzer?

### DISCUSSION THREAD

**Discussion by Summpot** [dotnet/roslyn#67359] [Q&A] [answered]
In general, an assembly cannot reference itself.
 However, for source generators, or analyzers, this is not the usual case. Unlike regular assemblies, people typically do not directly access the types and members of a source generator, and there is no actual dependency between the source generator and the assembly that references it.

So, can a source generator reference itself as an analyzer?
 It may sound strange, but there could be benefits to doing so, such as creating a source generator that generates code for the source generator, and being able to reference itself for consistency.

If there is a way to do this, please let me know.
 If not, could support for this be considered? Thank you!

**Comment by CyrusNajmabadi** [comment, marked as answer]
I would design things such that there was a common core analysis component.  Then the SG code and the DiagnosticAnalyzer code would both reference that analysis component.

**Reply by Summpot** [reply]
It sounds great, does this mean that people can combine different analysis components to analyze code generated by other source generators?
